lib/cable_ready/config.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "monitor"
+require "observer"
+require "singleton"
+
+module CableReady
+  # This class is a process level singleton shared by all threads: CableReady::Config.instance
+  class Config
+    include Singleton
+    include Observable
+
+    def initialize
+      @lock = Monitor.new
```

Comment:
I've just discovered that there is [MonitorMixin](https://ruby-doc.org/stdlib-2.6.1/libdoc/monitor/rdoc/MonitorMixin.html) which simplifies this a bit.

From https://www.ruby-forum.com/t/thread-synchronization-mutex-or-monitor/54365/9